polyguice-config/src/main/java/com/flipkart/polyguice/config/ApacheCommonsConfigProvider.java

```diff
@@ -45,36 +48,25 @@ public ApacheCommonsConfigProvider() {
         rootConfig = new CompositeConfiguration();
     }
 
-    public ApacheCommonsConfigProvider location(String loc) {
+    public ApacheCommonsConfigProvider location(String filePath) {
+        Class<? extends Configuration> clazz = selectConfigImpl(filePath);
         try {
-            if(loc.toLowerCase(Locale.getDefault()).endsWith(".properties")) {
-                PropertiesConfiguration config = new PropertiesConfiguration(new File(loc));
-                rootConfig.addConfiguration(config);
-                LOGGER.debug("properties configuration from {}", loc);
-            }
-            else if(loc.toLowerCase(Locale.getDefault()).endsWith(".xml")) {
-                XMLConfiguration config = new XMLConfiguration(new File(loc));
-                rootConfig.addConfiguration(config);
-                LOGGER.debug("xml configuration from {}", loc);
-            }
-            else if(loc.toLowerCase(Locale.getDefault()).endsWith(".json")) {
-                JsonConfiguration config = new JsonConfiguration(new File(loc));
-                rootConfig.addConfiguration(config);
-                LOGGER.debug("json configuration from {}", loc);
-            }
-            else if(loc.toLowerCase(Locale.getDefault()).endsWith(".yml")) {
-                YamlConfiguration config = new YamlConfiguration(loc);
-                rootConfig.addConfiguration(config);
-                LOGGER.debug("yaml configuration from {}", loc);
-            }
-            else if(loc.toLowerCase(Locale.getDefault()).endsWith(".yaml")) {
-                YamlConfiguration config = new YamlConfiguration(loc);
-                rootConfig.addConfiguration(config);
-                LOGGER.debug("yaml configuration from {}", loc);
+            if (clazz != null) {
+                this.rootConfig.addConfiguration(clazz.getConstructor(File.class).newInstance(new File(filePath)));
             }
+        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            throw new RuntimeException(e);
         }
-        catch(Exception exep) {
-            LOGGER.error("unable to load configuration from " + loc.toString(), exep);
+        return this;
+    }
+    public ApacheCommonsConfigProvider location(URL url) {
```

Comment:
Instead of having overloaded methods, one taking a URL and the other a string, why not have the original method's implementation extended to accept URL-like capabilities. So instead of taking a decision by looking at the path suffix (e.g. .properties, .yaml, etc.). also look at the prefix to determine load path. For example a path that looks like file:/somepath/filename.yaml will be loaded from the filepath, while a path of the form classpath:/somepath/filename.yaml will be loaded using the classpath. Default behavior, without a prefix will always assume a filepath.